src/plugins/data/common/query/to_expression_ast.ts

```diff
@@ -72,6 +72,24 @@ export async function queryStateToExpressionAst({
         throw new Error(`No data view found for index pattern ${idxPattern}`);
       }
     }
+
+    // esql query
+    if (mode === 'esql' && 'esql' in query) {
+      const idxPattern = getIndexPatternFromSQLQuery(query.esql);
+      const idsTitles = await dataViewsService.getIdsWithTitle();
+      const dataViewIdTitle = idsTitles.find(({ title }) => title === idxPattern);
```

Comment:
seems we can skip this part as we don't use the timefield in aggregateQueryToAst for esql

Reply:
That's true, but I would like to keep this API in place as support for this will most likely be added very soon one way or the other

Reply:
Valid point, removed it for now